compiler/plugins/target/ROCM/test/external_function_validation.mlir

```diff
@@ -0,0 +1,60 @@
+          %c0 = arith.constant 0 : index
+          %thread_id_x = gpu.thread_id  x
+          %thread_id_y = gpu.thread_id  y
+          %0 = hal.interface.binding.subspan layout(<bindings = [#hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">, #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">, #hal.pipeline.binding<storage_buffer, Indirect>], flags = Indirect>) binding(0) alignment(64) offset(%c0) flags("ReadOnly|Indirect") : memref<4096x4096xf16, strided<[4096, 1], offset: ?>>
+          %1 = hal.interface.binding.subspan layout(<bindings = [#hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">, #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">, #hal.pipeline.binding<storage_buffer, Indirect>], flags = Indirect>) binding(1) alignment(64) offset(%c0) flags("ReadOnly|Indirect") : memref<4096x4096xf16, strided<[4096, 1], offset: ?>>
+          %2 = vector.load %0[%thread_id_x, %thread_id_y] : memref<4096x4096xf16, strided<[4096, 1], offset: ?>>, vector<1xf16>
+          %3 = func.call @external_func(%2) : (vector<1xf16>) -> vector<1xf16>
+          vector.store %3, %1[%thread_id_x, %thread_id_y] : memref<4096x4096xf16, strided<[4096, 1], offset: ?>>, vector<1xf16>
+          return
+        }
+      }
+    }
+  }
+  util.func public @isolated_benchmark(%arg0: !hal.buffer_view, %arg1: !hal.buffer_view) -> !hal.buffer_view attributes {iree.abi.stub, iree.reflection = {iree.abi.declaration = "sync func @isolated_benchmark(%input0: tensor<4096x4096xf16>, %input1: tensor<4096x4096xf16>) -> (%output0: tensor<4096x4096xf16>)"}} {
+    %c33554432 = arith.constant 33554432 : index
+    %c0 = arith.constant 0 : index
+    %c4096 = arith.constant 4096 : index
+    %element_type_f16 = hal.element_type<f16> : i32
+    %dense_row_major = hal.encoding_type<dense_row_major> : i32
+    hal.buffer_view.assert<%arg0 : !hal.buffer_view> message("input0") shape([%c4096, %c4096]) type(%element_type_f16) encoding(%dense_row_major)
+    %0 = stream.tensor.import on(#hal.device.affinity<@__device_0>) %arg0 : !hal.buffer_view -> tensor<4096x4096xf16> in !stream.resource<external>{%c33554432}
+    hal.buffer_view.assert<%arg1 : !hal.buffer_view> message("input1") shape([%c4096, %c4096]) type(%element_type_f16) encoding(%dense_row_major)
+    %1 = stream.tensor.import on(#hal.device.affinity<@__device_0>) %arg1 : !hal.buffer_view -> tensor<4096x4096xf16> in !stream.resource<external>{%c33554432}
+    %result, %result_timepoint = stream.resource.alloca uninitialized on(#hal.device.affinity<@__device_0>) : !stream.resource<external>{%c33554432} => !stream.timepoint
+    %2 = stream.cmd.execute on(#hal.device.affinity<@__device_0>) await(%result_timepoint) => with(%0 as %arg2: !stream.resource<external>{%c33554432}, %1 as %arg3: !stream.resource<external>{%c33554432}, %result as %arg4: !stream.resource<external>{%c33554432}) {
+      stream.cmd.dispatch @test::@rocm_hsaco_fb::@test {
```

Comment:
Do you need this dispatch so that the executable doesn't get CSE'd? Can we just isolate the executable and call that pass for serializing executables? Why do we need to invoke the whole pipeline here?